### ISSUE TITLE
docker-machine ls stateTimeoutDuration is hardcoded to 10sec

### ISSUE THREAD

**Issue by wenlock** [docker/machine#2534]
Hi, I'm trying to find out if we have any specific reason why stateTimeoutDuration in docker-machine ls is hard coded to 10sec?  It would be nice to make this adjustable from the plugins or from a command line option.   Has any thoughts been given for that?

**Comment by nathanleclaire** [comment]
No particular reason, just hasn't been implemented.  My preference would probably be CLI option.

cc @docker/machine-maintainers WDYT?

**Comment by wenlock** [comment]
@nathanleclaire how about a cli option that can set it, or a way to get it from the plugin itself, because plugin developers should know a reasonable response time for their api's...

**Comment by nathanleclaire** [comment]
CLI option seems best to me.  I don't think it's really reasonable to configure that at a per-driver level.

**Comment by jeanlaurent** [comment]
Driver is one end of the tunnel, but timeout can really be affected from the "client side" too.

**Comment by jeanlaurent** [comment]
Is this a duplicate of #1696 ?

**Comment by wenlock** [comment]
@jeanlaurent @nathanleclaire this looks exactly like our issue as well.   Looks like @mattjmcnaughton last word on it appears to have been waiting for this refactor:
https://github.com/docker/machine/pull/1729/files#r39319122

guess his PR is way outdate, and needs another attempt.   Right?

**Comment by wenlock** [comment]
One thing c.Int no longer appears in the latest code.... how do you pull in new command options for new types?   finding it hard to follow...

**Comment by mattjmcnaughton** [comment]
@wenlock Sorry, I've had kind of a crazy semester at college trying to work on my thesis and haven't gotten too much of a chance to look at this. Yup, my pr is outdated - I think there have been some fairly large refactorings since then. Good luck!

**Comment by nathanleclaire** [comment]
Support for this is merged now :+1:

**Comment by wenlock** [comment]
wow, thats awesome, thanks!
